subsys/bluetooth/mesh/pb_gatt_srv.c

```diff
@@ -226,26 +226,27 @@ static size_t gatt_prov_adv_create(struct bt_data prov_sd[2])
 	sys_put_be16(prov->oob_info, prov_svc_data + 18);
 
 	if (!prov->uri) {
-		return 0;
+		goto dev_name;
 	}
 
 	uri_len = strlen(prov->uri);
 	if (uri_len > 29) {
 		/* There's no way to shorten an URI */
 		LOG_WRN("Too long URI to fit advertising packet");
-		return 0;
+		goto dev_name;
```

Comment:
This function will never return 0 anymore. It will return 1 if it enters one of these two if statements and 2 if the statements are false. Is that correct functionality? If so I do not have any other comments?
It will return 0 if CONFIG_BT_MESH_PB_GATT_USE_DEVICE_NAME is disabled and any of the if statements are true.

Reply:
Discussed in person. I belive this is correct functionality.